### PR TITLE
Bugfix for file system unresolvable UID/GID

### PR DESCRIPTION
**Pull request by taohe1012**
# Description
bugfix for filesystem with unresolvable GID/UID.
Issue: powerscale provider v1.2.0 resource Filesystem fails to create when UID/GID is not resolvable on cluster

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
|https://github.com/dell/dell-terraform-providers/issues/16 |

# ISSUE TYPE

- Bugfix Pull Request

##### RESOURCE OR DATASOURCE NAME
File System Resource

##### OUTPUT
All UT and AT for filesystem are passed:
image
image
image

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 80% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Unit tests
- [x] Acceptance tests

### CODE REVIEW

**Comment by forrestxia** [comment]
run e2e test

File: powerscale/helper/file_system_helper.go [hunk 1]
```diff
@@ -281,7 +281,149 @@ func BuildFilesystemDatasource(ctx context.Context, state *models.FileSystemData
...
+			ownerResType = types.StringValue(owner.GetType())
+		}
+		ownerTypeUnknown := false
+		if plan.Owner.Type.IsUnknown() {
+			ownerTypeUnknown = true
+			plan.Owner.Type = ownerResType
+		} else if !ownerResType.Equal(plan.Owner.Type) {
+			diags.AddWarning("File System ACL Owner type is invalid", fmt.Sprintf("File System ACL Owner type is invalid, actual type is '%s'", ownerResType.ValueString()))
+		}
+		if !plan.Owner.ID.IsUnknown() && plan.Owner.Name.IsUnknown() {
```
Comment (NavisJ) [inline powerscale/helper/file_system_helper.go:312]: The owner and group part have several conditions and may easily get confused. Can we add some comment to help understand?

Also please rebase the `main` branch

Comment (taohe1012) [reply powerscale/helper/file_system_helper.go]: thanks @NavisJ , rebased and comments added; 
if still has confusion in comments please let me known, thanks

**Comment by taohe1012** [comment]
run e2e test

**Review by NavisJ — APPROVED** [review]

**Pull request merged**